src/library/basetracksetfeature.cpp

```diff
@@ -0,0 +1,15 @@
+#include "library/basetracksetfeature.h"
+
+BaseTrackSetFeature::BaseTrackSetFeature(
+        Library* pLibrary,
+        UserSettingsPointer pConfig,
+        const QString& rootViewName)
+        : LibraryFeature(pLibrary, pConfig),
+          m_rootViewName(rootViewName) {
+}
+
+void BaseTrackSetFeature::activate() {
+    emit switchToView(m_rootViewName);
+    emit disableSearch();
```

Comment:
We're disabling searching for crates??

Reply:
Oh I see this is how it was before...

Reply:
yep, this PR is a pure refactor, no functional changes (except for this minimal menu order change which was made for consistency: https://github.com/mixxxdj/mixxx/pull/2621/commits/2e835ddc783e34c1fd0dbbfbd23aecb1d3f000bb)

Reply:
that's why I want to get it merged, further refactoring is not unlikely to introduce functional changes